### PR TITLE
side nav changes for SA

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
SA side navigaion changes

### Issues Resolved
Image 7-22-24 at 10 05 AM

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.ts [hunk 1]
```diff
@@ -43,6 +46,13 @@ export class SecurityAnalyticsPlugin
...
     { dataSourceManagement }: SecurityAnalyticsPluginSetupDeps
   ): SecurityAnalyticsPluginSetup {
+
+    const mountWrapper = async (params: AppMountParameters, redirect: string) => {
+      const { renderApp } = await import("./security_analytics_app");
+      const [coreStart, depsStart] = await core.getStartServices();
+      return renderApp(coreStart, params, redirect, depsStart, dataSourceManagement);
+    };
+
     core.application.register({
```
Comment (SuZhou-Joe) [inline public/plugin.ts:56]: Can we add workspaceAvailability for all the new features of security analytics.

Comment (riysaxen-amzn) [reply public/plugin.ts:56]: addressed

File: public/plugin.ts [hunk 2]
```diff
@@ -58,6 +68,133 @@ export class SecurityAnalyticsPlugin
...
+        order: 9080,
+        category: DEFAULT_APP_CATEGORIES.configure,
+        workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
+        mount: async (params: AppMountParameters) => {
+          return mountWrapper(params, ROUTES.LOG_TYPES);
+        },
+      });
+
+      const navlinks = [
+        { id: THREAT_ALERTS_NAV_ID },
```
Comment (SuZhou-Joe) [inline public/plugin.ts:178]:
```suggestion
        { id: THREAT_ALERTS_NAV_ID, showInAllNavGroup: true },
```

File: public/plugin.ts [hunk 3]
```diff
@@ -58,6 +68,133 @@ export class SecurityAnalyticsPlugin
...
+        { id: FINDINGS_NAV_ID },
```
Comment (SuZhou-Joe) [inline public/plugin.ts:179]:
```suggestion
        { id: FINDINGS_NAV_ID, showInAllNavGroup: true },
```

File: public/plugin.ts [hunk 4]
```diff
@@ -58,6 +68,133 @@ export class SecurityAnalyticsPlugin
...
+          id: INVESTIGATE_CATEGORY_ID,
+          category: DEFAULT_APP_CATEGORIES.investigate,
+          showInAllNavGroup: true
+        },
+        {
+          id: CONFIGURE_CATEGORY_ID,
+          category: DEFAULT_APP_CATEGORIES.configure,
+          showInAllNavGroup: true
+        },
+      ])
```
Comment (SuZhou-Joe) [inline public/plugin.ts:86]:
```suggestion
```
Seems useless

File: public/plugin.ts [hunk 5]
```diff
@@ -58,6 +68,133 @@ export class SecurityAnalyticsPlugin
...
+        { id: CORRELATIONS_NAV_ID },
+        { id: DETECTORS_NAV_ID },
+        { id: DETECTORS_RULE_NAV_ID, parent: DETECTORS_NAV_ID },
+        { id: CORRELATIONS_RULE_NAV_ID },
+        { id: THREAT_INTEL_NAV_ID},
+        { id: LOG_TYPES_NAV_ID }
+      ]
+
+      const navLinks = navlinks.map(item => ({
+        id: item.id,
```
Comment (SuZhou-Joe) [inline public/plugin.ts:177]:
```suggestion
        id: item.id,
        showInAllNavGroup: item. showInAllNavGroup,
```

File: public/plugin.ts [hunk 6]
```diff
@@ -58,6 +68,133 @@ export class SecurityAnalyticsPlugin
...
+        parentNavLinkId: item.parent ? item.parent : undefined,
+      }));
+
+      core.chrome.navGroup.addNavLinksToGroup(
+        DEFAULT_NAV_GROUPS['security-analytics'],
+        navLinks
+        );
+      }
```
Comment (SuZhou-Joe) [inline public/plugin.ts:185]:
```suggestion
      );
    }
```

File: public/utils/constants.ts [hunk 7]
```diff
@@ -19,6 +19,16 @@ export const PLUGIN_NAME = 'opensearch_security_analytics_dashboards';
...
+export const THREAT_ALERTS_NAV_ID = `threat_alerts`;
+export const FINDINGS_NAV_ID = `findings`;
+export const CORRELATIONS_NAV_ID = `correlations`;
+export const DETECTORS_NAV_ID = `detectors`;
+export const DETECTORS_RULE_NAV_ID = `detectors_rules`;
+export const CORRELATIONS_RULE_NAV_ID = `correlation_rules`;
+export const THREAT_INTEL_NAV_ID = `threat_intelligence`;
+export const LOG_TYPES_NAV_ID = `log_types`;
+export const CONFIGURE_CATEGORY_ID = `configure`;
+export const INVESTIGATE_CATEGORY_ID = `investigate`;
```
Comment (SuZhou-Joe) [inline public/utils/constants.ts:31]:
```suggestion
```
Not needed I think.

**Review by SuZhou-Joe — APPROVED** [review]

**Comment by riysaxen-amzn** [comment]
CI passing locally:

```

Test Suites: 36 passed, 36 total
Tests:       63 passed, 63 total
Snapshots:   37 passed, 37 total
Time:        95.477 s
Ran all test suites.

```

**Comment by riysaxen-amzn** [comment]
CI is failing due to operation canceled, it has nothing to do with the changes in this PR

**Review by AWSHurneyt — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**